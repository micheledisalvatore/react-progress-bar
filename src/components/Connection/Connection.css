.connection {
    width: calc(100% - 2.4em);
    height: 0.25em;
    background: #DBDBDB;
    position: absolute;
    bottom: 1.3em;
    left: calc(1.2em - 50%);
    z-index: 0;
}

.connection:after {
    content: '';
    background-color: #5131D7;
    height: 0.25em;
    left: 0;
    position: absolute;
    transition: width 500ms ease-in-out;
    width: 0;
    z-index: 2;
}

.connection--is-active:after {
    width: 100%;
    transition: width 500ms ease-in-out;
}