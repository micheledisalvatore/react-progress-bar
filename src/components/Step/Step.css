.step {
    flex-grow: 1;
    position: relative;
}

.step__dot {
    text-align: center;
    display: inline-block;
    position: absolute;
    top: -2.3em;
    left: 0;
}

.step__description {
    position: relative;
    letter-spacing: 1px;
    font-weight: bold;
    color: #aaa;
    top: -1em;
}

.step__description--is-active {
    color: #5131D7;
    transition: color 0s linear;
    transition-delay: 500ms;
}