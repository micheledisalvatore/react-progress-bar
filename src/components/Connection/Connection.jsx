import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './Connection.css';

export const Connection = ({ isActive }) => {
    const className = classNames('connection', {
        'connection--is-active': isActive,
    });

  return <div className={className} />;
};

Connection.propTypes = {
  isActive: PropTypes.bool,
};

Connection.defaultProps = {
  isActive: false,
};