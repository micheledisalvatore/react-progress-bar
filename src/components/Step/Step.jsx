import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import Connection from '../Connection';
import Dot from '../Dot';

import './Step.css';

export const Step = ({title, isFirst, isActive, isClickable, onClick, step}) => {
  const stepDescriptionClassName = classNames('step__description', {
    'step__description--is-active': isActive,
  });

  return (
    <div className="step">
      { !isFirst && <Connection isActive={isActive}/> }
      <div className={stepDescriptionClassName}>{title}</div>
      <Dot isActive={isActive} onClick={onClick} step={step} isClickable={isClickable} />
    </div>
  )
};

Step.propTypes = {
  isActive: PropTypes.bool,
  isFirst: PropTypes.bool,
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  step: PropTypes.oneOf([1, 2, 3, 4, 5]).isRequired,
};

Step.defaultProps = {
  isActive: false,
  isFirst: false,
};