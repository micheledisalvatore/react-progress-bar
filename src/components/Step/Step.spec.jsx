import React from 'react';
import { shallow } from 'enzyme';

import Connection from '../Connection';
import { Step } from './Step';

describe('Given a Step component', () => {
  describe('when it is rendered', () => {
    const TITLE = 'titleFoo';
    const STEP = 1;
    const onClickMock = jest.fn();
    const requiredProps = {
      title: TITLE,
      step: STEP,
      onClick: onClickMock,
    };
    const component = <Step {...requiredProps} />;
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(component);
    });

    afterEach(() => {
      onClickMock.mockReset();
    });

    it('should match the snapshot', () => {
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('should render a Connection component', () => {
      expect(wrapper.find(Connection)).toBePresent();
    });

    describe('and it is the first step', () => {
      beforeEach(() => {
        wrapper.setProps({
          isFirst: true,
        });
      });

      it('should NOT render a Connection component', () => {
        expect(wrapper.find(Connection)).not.toBePresent();
      });
    });
  });
});