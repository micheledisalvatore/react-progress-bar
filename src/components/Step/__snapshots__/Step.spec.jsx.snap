// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Step component when it is rendered should match the snapshot 1`] = `
<div
  className="step"
>
  <Unknown
    isActive={false}
  />
  <div
    className="step__description"
  >
    titleFoo
  </div>
  <Unknown
    isActive={false}
    onClick={[Function]}
    step={1}
  />
</div>
`;
